up-l1/mqtt_5.adoc

```diff
@@ -235,5 +235,43 @@ When MQTT5 (broker) is used for D2D communication, the topic shall consist of on
 ===== Sending a Message
 `UTransport::send(UMessage)` translates to the MQTT publish to topic `{UMessage.source.authority_name}/{UMessage.sink.authority_name}`
 
+== MQTT5 Connection handling
 
+=== Establishing a connection
+[.specitem,oft-sid="req~up-transport-mqtt5-establish-connection~1",oft-needs="impl,utest"]
+--
+To establish an MQTT connection to a broker it *MUST* be possible for a UTransport to configure the `cleanSession` and `sessionExpiry` settings.
+
+Those values have some important consequences summarized shortly:
+
+[cols="1,1,1"]
+|===
+| cleanSession | sessionExpiry | Consequence
+| true | 0 | Previous session data is deleted and state like messages, subscribtions etc. is lost if UTransport gets disconnected
+| true | t>0 | Previous session data is deleted and new session will buffer data for time `t` if connection is lost
+| false | t | Previous session including subscribtions, unreceived messages etc. is resumed and session will buffer data for time `t` if connection is lost
+|===
+
+Using a UTransport from the cloud side to connect to an MQTT broker we recommend to use `cleanSession=true` and `sessionExpiry=0` because we assume that there will always be at least one UTransport connected from the cloud which can handle messages so there is no need for buffering. 
+
+Establishing an mqtt5 UTransport from a device there is no general recommendation. Using `cleanSession=true` and `sessionExpiry=0` might be best for scaling since it reduces the load on the MQTT broker. Doing this one needs to take special care for offline devices. Using `cleanSession=true` and `sessionExpiry=t` might stress the broker but the broker helps when dealing with offline devices.
+
+--
+
+=== Reconnection
+[.specitem,oft-sid="req~up-transport-mqtt5-reconnection~1",oft-needs="impl,utest"]
+--
+If a connection is lost the UTransport *MUST* try to reconnect to the MQTT broker.  If on successfull reconnect the broker sends the `sessionPresent=false` flag then the UTransport *MUST* re-register all the previosly registred UListener. 
+--
+
+[.specitem,oft-sid="req~up-transport-mqtt5-reconnection-backoff~1",oft-needs="impl,utest"]
+--
+The client *MUST* implement a backoff strategy in order not to overload the MQTT broker. It *MUST* be possible to inject this strategy into the UTransport. The UTransport *MUST* implement the default strategy where the n+1-th reconnect is done after the n-th after a pause given by the following table:
```

Comment:
> It *MUST* be possible to inject this strategy into the UTransport.

What is that supposed to mean? Should it be possible to enable/disable the backoff, or parameterize it? If so, how? Unless we are more specific in this point I would propose to simply require the transport to implement the standard exponential backoff algorithm and use always use it ...

Reply:
Idea was to allow custom backoff strategies. I agree that just specifying the standard exponential backoff is good enough.

Reply:
https://github.com/eclipse-uprotocol/up-spec/commit/08f440905b2457fbcecb7a72f805131b1d7b9780